torch_geometric/nn/conv/lightgcn_conv.py

```diff
@@ -0,0 +1,63 @@
+from torch_geometric.typing import Adj
+
+import torch
+from torch import Tensor
+from torch_geometric.nn import MessagePassing
+
+
+class LightGCNConv(MessagePassing):
+    r"""The neighbor aggregation operator from the `"LightGCN: Simplifying and
+    Powering Graph Convolution Network for Recommendation"
+    <https://arxiv.org/abs/2002.02126#>`_ paper
+
+    .. math::
+        \textbf{e}_u^{(k+1)} =
+        \sum_{i \in N_u} \frac{1}{\sqrt{|N_u|} \sqrt{|N_i|}} \textbf{e}_i^{(k)}
+
+    .. math::
+        \textbf{e}_i^{(k+1)} =
+        \sum_{i \in N_i} \frac{1}{\sqrt{|N_i|} \sqrt{|N_u|}} \textbf{e}_u^{(k)}
+
+    Args:
+        in_channels (int): Size of each input sample, or :obj:`-1` to derive
+            the size from the first input(s) to the forward method.
+        out_channels (int): Size of each output sample.
+        num_users (int): Number of users for recommendation.
+        num_items (int): Number of items to recommend.
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
+    """
+    def __init__(self, in_channels: int, out_channels: int,
+                 num_users: int, num_items: int, **kwargs):
+        super(LightGCNConv, self).__init__(**kwargs)
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+
+        self.num_users = num_users
```

Comment:
My understanding is that `num_users` and `num_items` is not arbitrary, but is computed from `max(edge_index[0])+1`, `max(edge_index[1])+1`. Is my understanding correct? 
If so, we should consider inferring this from users or at least providing a clearer guideline here.

Reply:
Yes, your understanding is correct. Thanks for pointing this out. I've modified the code to infer `num_users` and `num_items` from `edge_index` as shown [here](https://github.com/KathyFeiyang/pytorch_geometric/blob/feat_lightgcn/torch_geometric/nn/conv/lg_conv.py#L35-L36).